docs/python-examples/fixtures/conftest.py

```diff
@@ -0,0 +1,111 @@
+import logging
+
+import pytest
+
+import ocs
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+@pytest.fixture(scope='class')
+def storage_class(request):
+    """
+    Storage class fixture
+
+    Returns:
+        StorageClass: object of storage class
+    """
+    sc_name = ocs.get_random_name('storage_class')
+    sc = ocs.StorageClass(sc_name)
+    logger.info(f"Creating storage class: {sc.name}")
+
+    def fin():
+        if not sc.is_deleted:
+            sc.delete()
+
+    request.addfinalizer(fin)
+    return sc_name
+
+
+@pytest.fixture(scope='class')
+def cls_pvc(request, storage_class):
+    """
+    PVC fixture
+
+    Returns:
+        PVC: object of pvc class
+    """
+    pvc_name = ocs.get_random_name('pvc')
+    pvc = ocs.PVC(pvc_name, storage_class)
+    logger.info(f"Creating pvc: {pvc.name}")
+
+    def fin():
+        if not pvc.is_deleted:
+            pvc.delete()
+
+    request.addfinalizer(fin)
+    return pvc
+
+
+@pytest.fixture(scope='class')
+def create_pvcs(request, storage_class):
```

Comment:
@andrewschoen the reason here was to really make sure the order for finalizer (forgot to mention this in comment), if I didn't specified it here you can see that storage class was deleted in different order that PVC in example here where you have some vimdiff output from two executions, on the left side it's with defined storage_class here in factory and on right side without.    https://paste.fedoraproject.org/paste/Gc8WLgbYBTZkkkcybZlK4Q .  

If we will use just storage class from test how we will make sure that the required resource will be deleted in right order? as we need to create in order . SC, PVC and delete in order PVC, SC for example and there will be more complex matrix I think for some resources. 

We need to make sure that our fixture factories will be able to be parametrized and we need to keep order for setup and delete the resources in right way.

Reply:
> If we will use just storage class from test how we will make sure that the required resource will be deleted in right order? as we need to create in order . SC, PVC and delete in order PVC, SC for example and there will be more complex matrix I think for some resources.

I don't think we should design our fixtures to cover both of these cases. There's really no reason to make things more complicated than they need to be. Either the pvc fixture can depend on the storage_class fixture as a dependency and handle cleanup itself, or it can require the test to depend on the storage_class fixture and pass that to the function the pvc fixture returned. In either case the resources are cleaned up through their respective fixture finalizers.

Reply:
@andrewschoen @clacroix12  As we need to keep the storage class fixture parameter in this factory to make proper order of cleaning I updated tests to use it like this.  Also made the factory as a function returning just 1 PVC, so if user needs more of them he should create them in some loop from the test or fixture like I have three_pvcs fixture.  Please take a look and tell me if it looks better now.